### PR TITLE
[Guild Week] Added generator for basic design tokens

### PR DESCRIPTION
**Pull request by Effanuel**
## Description
* My current guild week task is to [create a util for generating design tokens](https://jira.wixpress.com/browse/WOAUILIB-2550). Added a util method to `Colors` class, since that method depends mostly on other `Colors` class method.

## Changelog
* Added `generateTokens` method for generating design tokens based on chosen color.

### CODE REVIEW

File: src/style/__tests__/colors.spec.js [hunk 1]
```diff
@@ -123,4 +123,62 @@ describe('style/Colors', () => {
...
+      expect(uut.isDark(chosenColor)).toEqual(true);
+      expect(uut.generateTokens(chosenColor, 'light')).toEqual({
+        $backgroundPrimaryHeavy: chosenColor,
+        $backgroundPrimaryLight: '#FFFAFF',
+        $backgroundPrimaryMedium: '#FACCFA',
+        $iconPrimary: chosenColor,
+        $iconPrimaryLight: '#F16FF1',
+        $outlinePrimary: chosenColor,
+        $textPrimary: chosenColor
+      });
```
Comment (Effanuel) [inline src/style/__tests__/colors.spec.js:139]: These returned colors don't really match the ones in [figma](https://www.figma.com/file/Krv1pLl7kq2L52vMRwd498/Guidelines?node-id=11032%3A571101). I assume `generatePalette()` would need fixing here.

File: src/style/colors.ts [hunk 2]
```diff
@@ -199,10 +199,58 @@ export class Colors {
...
+  generateColorPalette = _.memoize((color: string): string[] => {
+    const palette = this.generatePalette(color);
     return this.shouldSupportDarkMode && Scheme.getSchemeType() === 'dark' ? _.reverse(palette) : palette;
   });
 
+  generateTokens(color: string, mode: 'light' | 'dark') {
+    return mode === 'light' ? this.generateLightModeTokens(color) : this.generateDarkModeTokens(color);
+  }
+
+  private generateLightModeTokens(color: string) {
```
Comment (Effanuel) [inline src/style/colors.ts:214]: `generateLightModeTokens` and `generateDarkModeTokens` methods are very similar, but I would rather have them separate, then to couple them

File: src/style/__tests__/colors.spec.js (hunk 1 above)
Comment (lidord-wix) [reply src/style/__tests__/colors.spec.js:139]: Actually I think the fix should be in the figma board, I'll take it with the relevant UX.

**Review by lidord-wix — COMMENTED** [review]
Looks very good! thank you for taking this task :) 

Consider adding another method to apply the new design tokens automatically (we can assume that most of the users that will use the `generateTokens()` method will load them right after using the `loadSchemes()` method, so we can offer a method that gets color and load the relevant light and dark design tokens).

**Comment by lidord-wix** [comment]
> Looks very good! thank you for taking this task :)
> 
> Consider adding another method to apply the new design tokens automatically (we can assume that most of the users that will use the `generateTokens()` method will load them right after using the `loadSchemes()` method, so we can offer a method that gets color and load the relevant light and dark design tokens).

@Effanuel I think you missed this comment :)

File: src/style/colors.ts [hunk 3]
```diff
@@ -43,6 +43,18 @@ export class Colors {
...
+   * Load light and dark schemes based on generated design tokens
+   * @param color - palette color
+   */
+  loadDesignTokens(color: string) {
+    Scheme.loadSchemes({
+      light: this.generateLightModeTokens(color),
+      dark: this.generateDarkModeTokens(color)
+    });
+    Object.assign(this, Scheme.getScheme());
+  }
```
Comment (lidord-wix) [inline src/style/colors.ts:56]:
```suggestion
    this.loadSchemes({
      light: this.generateLightModeTokens(color),
      dark: this.generateDarkModeTokens(color)
    });
  }
```

**Review by lidord-wix — APPROVED** [review]
Looks great!
see my last comment :)

**Pull request merged**